File: DNN Platform/Admin Modules/Dnn.Modules.ModuleCreator/Templates/VB/Settings - User Control/Settings.ascx.vb
#Region "Copyright"

' 
' Copyright (c) _YEAR_
' by _OWNER_
' 

#End Region

#Region "Using Statements"

Imports System
Imports DotNetNuke.Entities.Modules
Imports DotNetNuke.Services.Exceptions

#End Region

Namespace _OWNER_._MODULE_

    Partial Public Class Settings
        Inherits ModuleSettingsBase

#Region "Base Method Implementations"

        Public Overrides Sub LoadSettings()
            Try
                If Not Page.IsPostBack Then
                    txtField.Text = DirectCast(TabModuleSettings("field"), String)
                End If
            Catch exc As Exception
                ' Module failed to load
                Exceptions.ProcessModuleLoadException(Me, exc)
            End Try
        End Sub

        Public Overrides Sub UpdateSettings()
            Try
                ModuleController.Instance.UpdateTabModuleSetting(TabModuleId, "field", txtField.Text)
            Catch exc As Exception
                ' Module failed to load
                Exceptions.ProcessModuleLoadException(Me, exc)
            End Try
        End Sub

#End Region

    End Class

End Namespace

File: DNN Platform/Admin Modules/Dnn.Modules.ModuleCreator/Templates/C#/Settings - User Control/Settings.ascx.cs
// Licensed to the .NET Foundation under one or more agreements.
// The .NET Foundation licenses this file to you under the MIT license.
// See the LICENSE file in the project root for more information
#region Using Statements

using System;
using DotNetNuke.Entities.Modules;
using DotNetNuke.Services.Exceptions;

#endregion

namespace _OWNER_._MODULE_
{

    public partial class Settings : ModuleSettingsBase
    {

        #region Base Method Implementations

        public override void LoadSettings()
        {
            try
            {
                if (!Page.IsPostBack)
                {
                    txtField.Text = (string)TabModuleSettings["field"];
                }
            }
            catch (Exception exc) // Module failed to load
            {
                Exceptions.ProcessModuleLoadException(this, exc);
            }
        }

        public override void UpdateSettings()
        {
            try
            {
                ModuleController.Instance.UpdateTabModuleSetting(TabModuleId, "field", txtField.Text);
            }
            catch (Exception exc) // Module failed to load
            {
                Exceptions.ProcessModuleLoadException(this, exc);
            }
        }

        #endregion

    }

}

